crates/annuity/src/benchmarking.rs

```diff
@@ -1,33 +1,74 @@
 use super::*;
-use frame_benchmarking::{account, benchmarks_instance_pallet, impl_benchmark_test_suite};
+use frame_benchmarking::v2::*;
 use frame_support::{assert_ok, traits::Currency};
 use frame_system::RawOrigin;
 use sp_runtime::traits::One;
 
 // Pallets
 use crate::Pallet as Annuity;
 
-benchmarks_instance_pallet! {
-    withdraw_rewards {
-        let origin: T::AccountId = account("Origin", 0, 0);
-        assert_ok!(T::BlockRewardProvider::deposit_stake(&origin, One::one()));
+#[instance_benchmarks]
+pub mod benchmarks {
+    use super::*;
+
+    #[benchmark]
+    fn on_initialize() {
+        let caller = whitelisted_caller();
+        assert_ok!(T::BlockRewardProvider::deposit_stake(&caller, One::one()));
+        T::Currency::make_free_balance_be(&Annuity::<T, I>::account_id(), Annuity::<T, I>::min_reward_per_block());
+
+        #[block]
+        {
+            assert_ok!(Annuity::<T, I>::begin_block(1u32.into()));
+        }
+    }
+
+    #[benchmark]
+    fn withdraw_rewards() -> Result<(), BenchmarkError> {
+        let caller = whitelisted_caller();
+        assert_ok!(T::BlockRewardProvider::deposit_stake(&caller, One::one()));
+
+        // this only returns an error for `VaultAnnuity`,
+        // since calls are not enabled for that instance we
+        // can set the weight of this call to zero
```

Comment:
Should we just remove the `call` item from the runtime for `VaultAnnuity`? Then we don't need this `can_withdraw_reward`, right?

Reply:
There is no call item on the `VaultAnnuity` instance, benchmarks will run if listed in `define_benchmarks`.

Reply:
We still need `Call` for the root extrinsics but anyway would the benchmarks not still run for everything?